src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/GerritServer.java

```diff
@@ -914,14 +915,21 @@ private void rename(String newName) {
      * @param oldName the old name of the Gerrit server
      */
     private void changeSelectedServerInJobs(String oldName) {
-        for (AbstractProject job : PluginImpl.getConfiguredJobs_(oldName)) {
-            GerritTrigger trigger = (GerritTrigger)job.getTrigger(GerritTrigger.class);
+        for (Job job : PluginImpl.getConfiguredJobs_(oldName)) {
+
+            // TODO: find a way to handle add/remove of triggers
+            if (!(job instanceof AbstractProject)) {
+                return;
```

Comment:
@amuniz At the moment, nothing with happen for a Workflow Job i.e. add/remove of triggers is not supported for workflow as things stand. That's why I marked it with a `TODO`. We need to find a way of handling it for workflow. `AbstractProject` has methods for adding and removing triggers, but that's no help for a Workflow job.

Trigger `rename` might be fixable by finding the current trigger and renaming the server on it Vs  removing and readding the trigger (at the moment, `rename` is supported by removing the trigger and adding a new one). Not sure yet how `remove` will be supported (when a Gerrit server is removed we need to find all Jobs with triggers for that server and remove them).